cmp/grid/Grid.js

```diff
@@ -347,9 +347,17 @@ export class Grid extends Component {
             run: ([api, records]) => {
                 if (api) {
                     runInAction(() => {
+
+                        // Workaround for AG-2879.
+                        // pre-emptively dump rows with internal ag API to avoid expensive removals
+                        if (this.pendingDeletionCount(records, api) > 50) {
```

Comment:
Woops - I had typed a big comment here but lost it somehow.  

Discussed this number with Lee - I would have started with something more like 500-1000 - the idea being that the really slow behavior was happening when we replaced 50k records with another 50k, so it's starting with a 100k array and cutting it down in 50k passes.  Total time depends on count being removed as well as interim size, but in the end we don't need to be too fancy - just find a good balance.

In general the incremental updates we make (which we want to ensure are stable and fast) don't tend to remove many rows, so this is  a low-pressure choice either way.

Finally I would be up for moving this to a constant just to doc it there...